.github/scripts/setup-env.sh

```diff
@@ -81,7 +52,8 @@ fi
 echo '::endgroup::'
 
 echo '::group::Install PyTorch'
-pip install --progress-bar=off --pre torch --index-url="${PYTORCH_WHEEL_INDEX}"
+pip install --progress-bar=off light-the-torch
+ltt install --progress-bar=off --pytorch-computation-backend="${GPU_ARCH_TYPE}${GPU_ARCH_VERSION}" --pytorch-channel=nightly torch
```

Comment:
I've used this for simplicity for now. If we don't want a third-party tool here (full disclosure, I'm the author: https://github.com/pmeier/light-the-torch), we can also make it work with `pytorch_pkg_helpers`. This is available on PyPI, but seems to be maintained as part of the `test-infra` repository. It will look more complicated though, since that was made to print the content for a file that needs to be sourced afterwards. After that the `PIP_INSTALL_TORCH` variable is available, that does what it says on the tin.

Reply:
As it is used only for testing, I'm fine with it, but would be hesitant to rely on it in the long run, as I  guess it would  not work if one decides to throw `rocm` or something else to the equation.

Reply:
ROCm works just fine:

```
❯ ltt install --pytorch-computation-backend=rocm5.2 --pytorch-channel=stable torch
Collecting torch
  Downloading https://download.pytorch.org/whl/rocm5.2/torch-1.13.1%2Brocm5.2-cp38-cp38-linux_x86_64.whl (1342.9 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 0.0/1.3 GB 11.7 MB/s eta 0:01:54
ERROR: Operation cancelled by user
❯ ltt install --pytorch-computation-backend=rocm5.4.2 --pytorch-channel=nightly torch
Collecting torch
  Downloading https://download.pytorch.org/whl/nightly/rocm5.4.2/torch-2.1.0.dev20230312%2Brocm5.4.2-cp38-cp38-linux_x86_64.whl (1538.9 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 0.0/1.5 GB 11.4 MB/s eta 0:02:15
ERROR: Operation cancelled by user
```

What doesn't work yet, is the auto detection of ROCm. Meaning, if you are on a machine where the ROCm binaries are supported and simply invoke `ltt install torch`, it will pull CPU binaries. On the flip side, if your machine supports CUDA, it will automatically pull the right version:

```
❯ ltt install torch
Collecting torch
  Downloading https://download.pytorch.org/whl/cu117/torch-1.13.1%2Bcu117-cp38-cp38-linux_x86_64.whl (1801.8 MB)
```

> or something else

Happy to hear about concerns here. Note that I'm not pushing to keep `light-the-torch` in here, but am generally curious how I could improve it.